### PR TITLE
Address issue 891

### PR DESCRIPTION
**Pull request by keflavich**
See #891 

This PR fixes the order of arguments and adds regression tests.

### CODE REVIEW

**Comment by keflavich** [comment]
This unfortunately relies on python 3 syntax.  The only workaround is the ugly kwargs parsing approach... which I guess we'll have to use

**Comment by keflavich** [comment]
uh oh... the new LAMDA writer is causing appveyor failures?  https://ci.appveyor.com/project/Astropy/astroquery/build/1.0.520/job/mvlsvj9ibrd53ecv

**Comment by bsipocz** [comment]
Ahh, indeed. Having ``skip ci`` in the last commit of #887 tricked us believing it got fixed. 

I reported this false status report once to GH, but they referred me to report it with the ci-s. I'll try to pick up the threads and report it again.

**Comment by keflavich** [comment]
@bsipocz I confirmed that this solves #891.  Should we merge this, or wait until we've understood appveyor's issues better?

**Comment by bsipocz** [comment]
I think we should go ahead with this as the appveyor issue is clearly related to #887 but was overseen there. I'll open an issue for that, and suggest to ignore the appveyor 3.6 build for the time being.

File: astroquery/simbad/core.py [hunk 1]
```diff
@@ -465,12 +465,16 @@ def query_criteria_async(self, cache=True, *args, **kwargs):
...
             (e.g., {'otype':'SNR'} will be rendered as otype=SNR)
+        cache : bool
+            Cache the query?
 
         Returns
         -------
         response : `requests.Response`
             Response of the query from the server
         """
+        cache = kwargs.pop('cache') if 'cache' in kwargs else True
```
Comment (pllim) [inline astroquery/simbad/core.py]: A bit late to the discussion here but can't this simply be
```python
cache = kwargs.pop('cache', True)
```
?

Comment (bsipocz) [inline astroquery/simbad/core.py]: Yes, good point.

Comment (keflavich) [inline astroquery/simbad/core.py]: yep, that's a better idiom.

**Pull request merged**